From b80a85a36933209a42a9aae893241d8d998c4273 Mon Sep 17 00:00:00 2001
From: Huang Chaofan <chaofan.huang@nxp.com>
Date: Thu, 6 Sep 2018 18:12:39 +0800
Subject: [PATCH 4551/5242] MLK-19472 VPU: Adjust V4L2_MEMORY_DMABUF of
 Encoder driver for Android

commit  a3b72bc3b1de233f7548074d6490b5749eaf83dd from
https://source.codeaurora.org/external/imx/linux-imx.git

Adjust V4L2_MEMORY_DMABUF of Encoder driver for Android by changing
luma and chroma base address

Signed-off-by: Huang Chaofan <chaofan.huang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/mxc/vpu-encoder-b0/vpu_encoder_b0.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/mxc/vpu-encoder-b0/vpu_encoder_b0.c b/drivers/mxc/vpu-encoder-b0/vpu_encoder_b0.c
index 6b498db..8dc706c 100644
--- a/drivers/mxc/vpu-encoder-b0/vpu_encoder_b0.c
+++ b/drivers/mxc/vpu-encoder-b0/vpu_encoder_b0.c
@@ -900,9 +900,9 @@ static bool update_yuv_addr(struct vpu_ctx *ctx, u_int32 uStrIdx)
 		vpu_dbg(LVL_INFO, "\n");
  #endif
 		pphy_address = (u_int32 *)vb2_plane_cookie(p_data_req->vb2_buf, 0);
-		pParamYuvBuffDesc->uLumaBase = *pphy_address;
+		pParamYuvBuffDesc->uLumaBase = *pphy_address + p_data_req->vb2_buf->planes[0].data_offset;
 		pphy_address = (u_int32 *)vb2_plane_cookie(p_data_req->vb2_buf, 1);
-		pParamYuvBuffDesc->uChromaBase = *pphy_address;
+		pParamYuvBuffDesc->uChromaBase = *pphy_address + p_data_req->vb2_buf->planes[1].data_offset;
     /* Not sure what the test should be here for a valid frame return from vb2_plane_cookie */
 		if (pParamYuvBuffDesc->uLumaBase != 0)
 			bGotAFrame = TRUE;
-- 
1.7.9.5

